### DISCUSSION TITLE
TS1400D

### DISCUSSION THREAD

**Discussion by redkurn** [1000001101000/Debian_on_Buffalo#103] [Q&A] [answered]
java -jar acp_commander.jar -t 192.168.0.18 -o
A SocketTimeoutException usually indicates bad firewall settings.
Check especially for *UDP* port 22936 and make sure that the connection to your LS is working.
ERROR: Exception: SocketTimeoutException (Receive timed out) [ACP Send/Receive (Packet:80A0 = SPECIAL_CMD_EN_ONECMD)]

I'm not sure which firewall is the issue, I've made an exception on windows firewall and on my router enabled forwarding, but still get this error.

Was on firmware 1.67, but rolled back to 1.60 using the debug menu on the updater.

Issuing
java -jar acp_commander.jar -f -t 192.168.0.18
is the only way acp_commander even sees my terastation and trying to change the boot files and installing with the stock firmware tool doesn't work at all.

I can get firmware 1.40 if this is an issue with the firmware blocking the port, but other than that I have been trying to work this out for a few days with no success.

**Comment by redkurn** [comment, marked as answer]
Found the solution, not sure why it didn't show up when I skimmed the posts...
https://github.com/1000001101000/Debian_on_Buffalo/issues/34

Disable windows firewall, still doesn't see the nas, but the commands are working otherwise.

**Comment by redkurn** [comment]
got it to install, but could only telnet in, ssh was not running.
not sure what went wrong, but instead of following the guide to restore to stock, i just checked all the boxes in the modified TSUpdater and it is formatting the drives and installing the 1.67 firmware.

will have to try again, need to get some sleep.

**Reply by 1000001101000** [reply]
Good to hear from you!

What are you trying to accomplish?

**Reply by redkurn** [reply]
I was trying to get a standard boot partition size (512mb -> 1gb) and at least 4gb swap with the remainder available for debian.

I stayed up and restarted the install from stock and now I have a working 5.1gb boot 4.1gb swap and the rest of the array running debian.
I hadn't tried using raid0 on linux previously, maybe I did it wrong or wiping the drives completely causes a boot error or lack of ssh service?

**Reply by 1000001101000** [reply]
When you complete the debian install the only service that runs is ssh. If acp_commander or telnet work you’ve still got the stock firmware at that point.

 In any case it sounds like you’re making progress. Let me know if you need help.

**Reply by redkurn** [reply]
I'm going to stick with what I have working at the moment, I would suggest a partitioning guide that works for these little boxes though.
Twonky is just horrible that comes installed and this as a little plex server will be amazing.

Thank you for putting all this together, How can I set the power led to amber when it is fully booted?

**Reply by 1000001101000** [reply]
Glad you asked!

This device has an odd mix. Some LEDs can be controlled via their entries in /sys/class/leds. That LED in particular is controlled by the onboard microcontroller. 

I've create a Python library for sending commands to the microcontroller which allows you to control the other LEDs as well as other things like the buzzer. The repo includes and example script for the ts1400. You ca find that here:
https://github.com/1000001101000/Python_buffalo_libmicon

**Reply by redkurn** [reply]
Apparently it can't run plex or any DNLA server, they all crash.
Kind of sidelined using this and set up a older pc to do the job, this project is awesome though.

**Reply by 1000001101000** [reply]
Can you provide more information of what you tried and what happened? I'm not experienced with plex but I'd be interested in taking a look .... I thought others had done DLNA successfully

**Reply by redkurn** [reply]
I installed the latest arm7 deb, I thought others did as well, but I haven't found anyone using debian on buffalo on a ts1400d.

**Reply by 1000001101000** [reply]
hmm, plex isn't what I thought it was. I took a quick look at it the Debian 8+ requirement makes me wonder since Debian has changed quite a bit between 8 and 10 but haven't tried it yet.

I did install minidlna quick and was able to play an mp3 with it. This was on a TS3400D which is similar but somewhat different, I can try with my TS1400D when I get a chance to set it up if that would be helpful.

**Reply by redkurn** [reply]
If you can get it to work properly and pass on how, that would be awesome!
I used to know my way around linux a bit better, but I haven't used Debian based in a long while, much less any distro really.
Last I was using had been Arch based and I never really learned enough command line or the ins and outs of doing things other than what I needed to get the OS up and running, while maintaining it.

I did learn how to build git and some other things, but figuring out what is wrong with Plex is way over my head.
